app/scripts/modules/amazon/src/domain/IAmazonLaunchTemplate.ts

```diff
@@ -0,0 +1,87 @@
+import { IBlockDeviceMapping } from './IAmazonBlockDeviceMapping';
+
+export interface IIamInstanceProfile {
+  arn?: string;
+  name: string;
+}
+
+export interface ICpuOptions {
+  coreCount?: number;
+  threadsPerCore?: number;
+}
+
+export interface IElasticGpuSpecification {
+  type?: string;
+}
+
+export interface IElasticInterfaceAccelerator {
+  count?: number;
+  type?: string;
+}
+
+export interface ISpotMarketOptions {
+  blockDurationMinutes?: number;
+  instanceInterruptionBehavior?: string;
+  maxPrice?: string;
+  spotInstanceType?: 'one-time' | 'persistent';
+  validUntil?: string;
+}
+
+export interface IInstanceMarketOptions {
+  marketType?: string;
+  spotOptions?: ISpotMarketOptions;
+}
+
+export interface ILicenseConfig {
+  licenseConfigurationArn?: string;
+}
+
+export interface IMetadataOptions {
+  httpEndpoint?: 'disabled' | 'enabled';
+  httpPutResponseHopLimit?: number;
+  httpsTokens?: 'required' | 'optional';
+  state?: 'pending' | 'applied';
+}
+
+export interface ITagSpecification {
+  resourceType?: string;
+  tagSet?: Array<{
+    [key: string]: string;
+  }>;
+}
+
+export interface ILaunchTemplateData {
+  [attribute: string]: any;
```

Comment:
It's best to avoid index types if possible because it makes type checking super loose.  I looked up this object type in AWS docs and it doesn't seem to suggest it can contain arbitrary keys.  Can you explain why this is needed?

Reply:
I definitely agree with you on the fact that it makes type checking loose. I would also say that we do not 'need' this. My think is that this type is dependent on a model that is still having features added to it. I felt like this helped us enforce stronger types than we currently have, while not being too strongly coupled to aws changes. 

The specific attributes are important to Spinnaker, and the rest could evolve over time. If they are important we can specifically include them as needed and if not we do not have to worry about backwards compatibility.